src/Monolog/Logger.php

```diff
@@ -176,6 +176,25 @@ public function popHandler()
     }
 
     /**
+     * Set handlers, removing all existing ones.
+     * Falsey values will be ignored, and if a map is passed, keys will be ignored.
+     *
+     * @param array $handlers All elements must be of type HandlerInterface
+     * @return $this
+     */
+    public function setHandlers(array $handlers)
+    {
+        $this->handlers = array();
+        foreach ($handlers as $handler) {
+            if ($handler) {
+                $this->pushHandler($handler);
```

Comment:
I think the foreach should do an array_reverse on $handlers because otherwise you are essentially flipping the order if I call `setHandlers([a, b])` then `b` will be called first then `a`, which is not really expected IMO.